builtin/audit/socket/backend.go

```diff
@@ -0,0 +1,167 @@
+package socket
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/vault/audit"
+	"github.com/hashicorp/vault/helper/duration"
+	"github.com/hashicorp/vault/logical"
+)
+
+func Factory(conf *audit.BackendConfig) (audit.Backend, error) {
+	if conf.Salt == nil {
+		return nil, fmt.Errorf("nil salt passed in")
+	}
+
+	address, ok := conf.Config["address"]
+	if !ok {
+		return nil, fmt.Errorf("address is required")
+	}
+
+	socketType, ok := conf.Config["socket_type"]
+	if !ok {
+		socketType = "tcp"
+	}
+
+	writeDeadline, ok := conf.Config["write_deadline"]
```

Comment:
Let's call this `write_timeout`. Even though it doesn't match the Go parameter, I think it's a little more understandable/intuitive for the average end user, since they're not actually specifying a deadline.